CMakeLists.txt

```diff
@@ -44,12 +44,25 @@ find_package(LAPACK REQUIRED)
 
 # Python required for application testing
 include(cmake/python3.cmake)
-find_python_module(numpy)
-find_python_module(pytest)
+
+find_package(Python3 3.7 COMPONENTS Interpreter Development)
+if(Python3_FOUND)
+    message("-- Python 3 interpreter version: " ${Python3_VERSION})
+else()
+    message("-- Python 3 interpreter not found")
+endif()
 
 # Enable ctest
 enable_testing()
 
+# Application test python module
+# Note, pygreenx dependencies should be handled pygreenx
+# pygreenx version specified in its setup.cfg
+find_python_module(pygreenx VERSION 0.0.1)
+if (NOT pygreenx_FOUND)
+    message("-- pygreenx is required for application testing. To install it, cd <PROJECTROOT>/python and run `pip install -e .`")
+endif ()
+
```

Comment:
I introduced these changes yesterday. They should not be showing in the diff. 

Furthermore, I just pulled main and I see you've been pushing to it. This is going to create a total mess.

I'm going to force revert main, and prevent developers pushing directly to it (I thought I'd already enabled that).

Reply:
I can't pull changes into main because som branches have been messed up. 

cannot lock ref 'refs/remotes/origin/JOSS/readme_and_test': 'refs/remotes/origin/JOSS' exists; cannot create 'refs/remotes/origin/JOSS/readme_and_test' From https://github.com/nomad-coe/greenX ! [new branch] JOSS/readme_and_test -> origin/JOSS/readme_and_test (unable to update local ref)